modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -1659,13 +1659,17 @@ def test_reset_index_with_named_index(
     df_equals(modin_df, pandas_df)
     if test_async_reset_index:
         # The change in index is not automatically handled by Modin. See #3941.
-        modin_df._query_compiler._modin_frame._propagate_index_objs(axis=0)
+        modin_df.index = modin_df.index
+        modin_df._to_pandas()
```

Comment:
Why do we need to reassign the index and call ._to_pandas()?

Reply:
This is a replacement for `_propagate_index_objs`. And it also works for HDK.